AndroidSDKTests/src/test/java/com/leanplum/LeanplumTest.java

```diff
@@ -1671,4 +1671,63 @@ public void variablesChanged() {
 
     assertEquals(4, VarCache.variants().size());
   }
+
+  @Test
+  public void testStartAndVariableChangeCallBacksForOffline() throws Exception {
+    final Semaphore semaphore = new Semaphore(1);
+    semaphore.acquire();
+
+    //Offline Mode.
+     ResponseHelper.seedResponseNull("/responses/simple_start_response.json");
+
+    // Expected request params.
+    final HashMap<String, Object> expectedRequestParams = CollectionUtil.newHashMap(
+        "city", "(detect)",
+        "country", "(detect)",
+        "location", "(detect)",
+        "region", "(detect)",
+        "locale", "en_US"
+    );
+
+    // Validate request.
+    // Validate request.
+    RequestHelper.addRequestHandler(new RequestHelper.RequestHandler() {
+      @Override
+      public void onRequest(String httpMethod, String apiMethod, Map<String, Object> params) {
+        assertEquals(Constants.Methods.START, apiMethod);
+        assertTrue(params.keySet().containsAll(expectedRequestParams.keySet()));
+        assertTrue(params.values().containsAll(expectedRequestParams.values()));
+      }
+    });
+
+
+    Leanplum.start(mContext, new StartCallback() {
+      @Override
+      public void onResponse(boolean success) {
+        assertTrue(success);
+        semaphore.release();
+      }
+    });
+    assertTrue(Leanplum.hasStarted());
+
+    semaphore.acquire();
+
+    // Seed getVars response.
```

Comment:
please could we split up these 2 responses into their own test files? It'd be great to keep the tests testing as little individually as possible so that when something breaks, we don't have to parse the test to understand what happened

Reply:
done